File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s)%2 ==1:
            return False
        closeToOpen = { ")" : "(", "]" : "[", "}" : "{" }
        stack = []
        for ch in s:
            if ch in closeToOpen.values():
                stack.append(ch)
            elif ch in closeToOpen:
                if not stack or closeToOpen[ch] != stack[-1]:
                    return False
                stack.pop()
        return True if not stack else False
                
                
                    

        

File: Data Structures & Algorithms/car-fleet/submission-2.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        Map<Integer,Integer> map = new HashMap<>();

        for(int i =0; i<position.length;i++){
            map.put(position[i],speed[i]);
        }
        Deque<Double> stack = new ArrayDeque<>();
        
        Arrays.sort(position);

        for(int i =position.length-1;i>=0;i--){
            if(stack.isEmpty()){
                double time = (double)(target -  position[i])/map.get(position[i]);
                stack.push(time);
            }else{
                double time = (double)(target -  position[i])/map.get(position[i]);
                double top = stack.peek();
                if(time>top){
                    stack.push(time);

                }
            }
            
        }
        return stack.size();




        
    }
}
